RemoteImageClassifierTask: processing stopped by "HTTP Status Code: 400 - Nenhuma imagem válida encontrada no arquivo ZIP"

This happened when processing 3 UFDR files. It is reproducible (checked running the same command 3 times). I had to resume the processing to complete. Changing the code to ignore `eHTTP.getStatusCode() == HttpStatus.SC_BAD_REQUEST` also worked. 

```
2026-05-09 04:44:25	[ERROR]	[engine.task.RemoteImageClassifierTask]			ClassificationFail::HttpStatusNotOK: HTTP Status Code: 400 - Response:
{"detail":"Nenhuma imagem válida encontrada no arquivo ZIP"}
2026-05-09 04:44:25	[INFO]	[engine.core.Worker]			Worker-24 interrupted on [queue-end] (null bytes)
2026-05-09 04:44:25	[INFO]	[engine.task.RemoteImageClassifierTask]			Send ZIP file #4499 (files: 1)
2026-05-09 04:44:25	[INFO]	[engine.task.RemoteImageClassifierTask]			Send ZIP file #4500 (files: 1)
2026-05-09 04:44:25	[INFO]	[jep.Jep]			[FaceRecognitionTask] Error loading image xxxxxx.ufdr/00008140-xxxxxxx_files_partial-a.zip/private/var/mobile/Containers/Shared/AppGroup/7F375402-DDA3-443B-8A84-038269BE96A4/ChatStorage.sqlite>>Carved-11939491.jpg (7878 bytes)
java.lang.Exception: Worker-24 Error while processing [queue-end] (nullbytes)
	at iped.engine.core.Worker.process(Worker.java:188)
	at iped.engine.core.Worker.run(Worker.java:293)
Caused by: java.lang.RuntimeException: HTTP Status Not OK on 'https://10.61.86.148:30603/new/zip': HTTP Status Code: 400 - Response:
{"detail":"Nenhuma imagem válida encontrada no arquivo ZIP"}
	at iped.engine.task.RemoteImageClassifierTask.sendBatchedFiles(RemoteImageClassifierTask.java:667)
	at iped.engine.task.RemoteImageClassifierTask.process(RemoteImageClassifierTask.java:728)
	at iped.engine.task.AbstractTask.processMonitorTimeout(AbstractTask.java:282)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:197)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.EmbeddedDiskProcessTask.sendToNextTask(EmbeddedDiskProcessTask.java:116)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:210)
	at iped.engine.core.Worker.process(Worker.java:179)
	... 1 more
Caused by: HTTP Status Code: 400 - Response:
{"detail":"Nenhuma imagem válida encontrada no arquivo ZIP"} HTTP Status Code: 400 - Response: {"detail":"Nenhuma imagem válida encontrada no arquivo ZIP"}
	at iped.engine.task.RemoteImageClassifierTask.sendZipFile(RemoteImageClassifierTask.java:637)
	at iped.engine.task.RemoteImageClassifierTask.sendBatchedFiles(RemoteImageClassifierTask.java:647)
	... 62 more
2026-05-09 04:44:26	[INFO]	[engine.core.Worker]			Worker-0 interrupted on [queue-end] (null bytes)
2026-05-09 04:45:52	[INFO]	[zaxxer.hikari.HikariDataSource]			HikariPool-1 - Shutdown initiated...
2026-05-09 04:45:52	[INFO]	[zaxxer.hikari.HikariDataSource]			HikariPool-1 - Shutdown completed.
```

--- a/iped-engine/src/main/java/iped/engine/data/Item.java
+++ b/iped-engine/src/main/java/iped/engine/data/Item.java
@@ -53,6 +53,7 @@ import iped.utils.ByteArrayImageInputStream;
 import iped.utils.EmptyInputStream;
 import iped.utils.HashValue;
 import iped.utils.IOUtil;
+import iped.utils.ImageUtil;
 import iped.utils.LimitedSeekableInputStream;
 import iped.utils.SeekableByteChannelImpl;
 import iped.utils.SeekableFileInputStream;
@@ -1358,7 +1359,13 @@ public class Item implements IItem {
     }
 
     public void setThumb(byte[] thumb) {
-        this.thumb = thumb;
+        if (ImageUtil.isImageValid(thumb)) {
+            this.thumb = thumb;
+        } else {
+            // Reject the invalid image data
+            // zero size thumb means thumb error
+            this.thumb = new byte[0];
+        }
     }
 
     public ISeekableInputStreamFactory getInputStreamFactory() {

--- a/iped-utils/src/main/java/iped/utils/ImageUtil.java
+++ b/iped-utils/src/main/java/iped/utils/ImageUtil.java
@@ -12,6 +12,7 @@ import java.awt.image.ColorConvertOp;
 import java.awt.image.ColorModel;
 import java.awt.image.DataBufferByte;
 import java.awt.image.WritableRaster;
+import java.io.ByteArrayInputStream;
 import java.io.ByteArrayOutputStream;
 import java.io.File;
 import java.io.FileInputStream;
@@ -837,5 +838,27 @@ public class ImageUtil {
         WritableRaster raster = source.copyData(null);
 
         return new BufferedImage(cm, raster, isAlphaPremultiplied, null);
+    }
+
+    /**
+     * Checks if a given byte array represents a structurally valid image.
+     * 
+     * @param imageData The byte array of the image/thumbnail.
+     * @return true if the image is valid and can be decoded, false otherwise.
+     */
+    public static boolean isImageValid(byte[] imageData) {
+        if (imageData == null || imageData.length < 10) {
+            return false;
+        }
+
+        try (InputStream is = new ByteArrayInputStream(imageData)) {
+            // ImageIO.read returns a BufferedImage if successful,
+            // or null if no registered ImageReader claims to be able to read the stream.
+            return ImageIO.read(is) != null;
+        } catch (Exception e) {
+            // Any parsing error (like unexpected EOF from carved fragments) means it's
+            // structurally invalid
+            return false;
+        }
     }
 }

